tools/bootstrap_commitlogs/main/main.go

```diff
@@ -0,0 +1,233 @@
+	shardsArg               = flag.String("shards", "", "Shards - set comma separated list of shards")
+	debugListenAddressArg   = flag.String("debug-listen-address", "", "Debug listen address - if set will expose pprof, i.e. ':8080'")
+	currentUnixTimestampArg = flag.String("current-unix-timestamp", currentUnixTimestampString(), "Current unix timestamp - If set will perform the bootstrap as if this was the current time, defaults to current time")
+)
+
+func currentUnixTimestampString() string {
+	unixTs := time.Now().Unix()
+	return strconv.Itoa(int(unixTs))
+}
+
+func main() {
+	flag.Parse()
+	if *pathPrefixArg == "" ||
+		*namespaceArg == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	var (
+		pathPrefix           = *pathPrefixArg
+		namespaceStr         = *namespaceArg
+		blockSize            = *blockSizeArg
+		flushSize            = *flushSizeArg
+		bootstrapRetention   = *bootstrapRetentionArg
+		shardsCount          = *shardsCountArg
+		shards               = *shardsArg
+		debugListenAddress   = *debugListenAddressArg
+		currentUnixTimestamp = *currentUnixTimestampArg
+	)
+
+	log := xlog.NewLogger(os.Stderr)
+
+	if debugListenAddress != "" {
+		go func() {
+			log.Infof("starting debug listen server at '%s'\n", debugListenAddress)
+			err := http.ListenAndServe(debugListenAddress, http.DefaultServeMux)
+			if err != nil {
+				log.Fatalf("could not start debug listen server at '%s': %v", debugListenAddress, err)
+			}
+		}()
+	}
+
+	shardTimeRanges := result.ShardTimeRanges{}
+
+	currentUnixTimestampParsed, err := strconv.Atoi(currentUnixTimestamp)
+	if err != nil {
+		log.Fatalf("could not parse unix timestamp: '%s': %v", currentUnixTimestamp, err)
+	}
+
+	now := time.Unix(int64(currentUnixTimestampParsed), 0)
+	// Round current time down to nearest blocksize (2h) and then decrease blocksize (2h)
+	startInclusive := now.Truncate(blockSize).Add(-blockSize)
+	// Round current time down to nearest blocksize (2h) and then add blocksize (2h)
+	endExclusive := now.Truncate(blockSize).Add(blockSize * 2)
+
+	// Ony used for logging
+	var shardsAll []uint32
+
+	// Handle commda-delimited shard list 1,3,5, etc
+	if strings.TrimSpace(shards) != "" {
+		for _, shard := range strings.Split(shards, ",") {
+			shard = strings.TrimSpace(shard)
+			if shard == "" {
+				log.Fatalf("Invalid shard list: '%s'", shards)
+			}
+			value, err := strconv.Atoi(shard)
+			if err != nil {
+				log.Fatalf("could not parse shard '%s': %v", shard, err)
+			}
+			rng := xtime.Range{Start: startInclusive, End: endExclusive}
+			shardTimeRanges[uint32(value)] = xtime.NewRanges().AddRange(rng)
+			shardsAll = append(shardsAll, uint32(value))
+		}
+		// Or just handled up to N (shard-count) shards
+	} else if shardsCount > 0 {
+		for i := uint32(0); i < uint32(shardsCount); i++ {
+			rng := xtime.Range{Start: startInclusive, End: endExclusive}
+			shardTimeRanges[i] = xtime.NewRanges().AddRange(rng)
+			shardsAll = append(shardsAll, i)
+		}
+	} else {
+		flag.Usage()
```

Comment:
might be better to tell the user that either shardCount or the comma list has to be set, and then print usage.

Reply:
done